gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/CopyableFile.java

```diff
@@ -354,6 +359,36 @@ public static OwnerAndPermission resolveReplicatedOwnerAndPermission(FileSystem
     return ownerAndPermissions;
   }
 
+  public static Map<String, OwnerAndPermission> resolveReplicatedAncestorOwnerAndPermissionsRecursively(FileSystem sourceFs, Path fromPath,
+      Path toPath, CopyConfiguration copyConfiguration) throws IOException {
+
+    Map<String, OwnerAndPermission> ownerAndPermissions = Maps.newHashMap();
+
+    // We only pass directories to this method anyways. Those directories themselves need permissions set.
```

Comment:
Should we add one check if you have assumption here?

Reply:
fromPath is a dir, because it is a .getParent() of some path. toPath is a calculated destination path, which may not exist yet, so cannot be tested to be a dir

Reply:
yeah I know now you only path dir to this method now, just want to add some check inside this method itself to make sure others will not using this method in a wrong way